app_legacy/Helpers/database/user.php

```diff
@@ -3,27 +3,16 @@
 use LegacyApp\Community\Enums\ClaimStatus;
 use LegacyApp\Community\Enums\TicketState;
 use LegacyApp\Site\Enums\Permissions;
+use LegacyApp\Site\Models\User;
 
-function GetUserData($user): ?array
+function GetUserData(string $username): ?array
 {
-    sanitize_sql_inputs($user);
-
-    $query = "SELECT * FROM UserAccounts WHERE User='$user'";
-    $dbResult = s_mysql_query($query);
-
-    if (!$dbResult || mysqli_num_rows($dbResult) != 1) {
-        log_sql_fail();
-
-        // failed: Achievement $id doesn't exist!
-        return null;
-    }
-
-    return mysqli_fetch_assoc($dbResult);
+    return User::find($username)?->toArray();
```

Comment:
This doesn't seem to work.
```
$ curl "localhost/dorequest.php?r=awardachievement&u=Jamiras&a=40403&h=1&m=8e3630186e35d477231bf8fd50e54cdd&t=xxxxxxxxxxxxxxxx"
{"Success":false,"Error":"User data cannot be found for Jamiras","Score":84514,"SoftcoreScore":1100,"AchievementID":40403}
```
Changing it to `firstWhere('User', $username)` does.

Reply:
Good catch, thanks! I missed that the username is not the primary key there - `find()` only works with the primary key.

I'll fix that and check that i did not add this mistake anywhere else.

Reply:
Fixed in https://github.com/RetroAchievements/RAWeb/pull/1393/commits/c534351ad4379613b3f5c1e224771c680108b453